### PR TITLE
fix(snowflake): ensure that memtables are translated correctly

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

**Comment by krzysztof-kwitt** [comment]
This is still a draft as I need to run full tests.

File: ibis/backends/snowflake/__init__.py [hunk 1]
```diff
@@ -74,9 +75,21 @@ class SnowflakeExprTranslator(AlchemyExprTranslator):
     supports_unnest_in_select = False
 
 
+class SnowflakeTableSetFormatter(_AlchemyTableSetFormatter):
+    def _format_in_memory_table(self, op, ref_op, translator):
+        columns = translator._schema_to_sqlalchemy_columns(ref_op.schema)
+        rows = list(ref_op.data.to_frame().itertuples(index=False))
+        pos_columns = [
+            sa.column(f"${idx + 1}").alias(d)
+            for idx, d in enumerate(ref_op.schema.names)
```
Comment (cpcloud) [inline ibis/backends/snowflake/__init__.py:84]:
```suggestion
            sa.column(f"${idx}").alias(d)
            for idx, d in enumerate(ref_op.schema.names, start=1)
```

**Review by cpcloud — APPROVED** [review]
Small comment, otherwise LGTM!

**Comment by krzysztof-kwitt** [comment]
```
$ pytest -m snowflake ibis/backends/tests --numprocesses 6 --dist=loadgroup
========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.10.7, pytest-7.2.1, pluggy-1.0.0
Using --randomly-seed=3362120479
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /ibis/ibis, configfile: pyproject.toml
plugins: snapshot-0.9.0, xdist-3.1.0, mock-3.10.0, clarity-1.0.1, randomly-3.12.0, profiling-1.7.0, cov-4.0.0, repeat-0.9.1, benchmark-4.0.0, hypothesis-6.65.2
gw0 [877] / gw1 [877] / gw2 [877] / gw3 [877] / gw4 [877] / gw5 [877]
.s......x...................x..x..............x..........................x..............x..............................xx..x.xx........x........xxxxxx..x.x.x.........x....x.......xx..xxxx.xx.... [ 22%]
x..x...x...xx........xxx.x..x..x.x..x........xx.xx..x........x.x....x...xx.xxxxxxxxxxx.xx..x..x.x......x..................x........................x............x............x..........x......... [ 44%]
....x.x..................x.x.x.....x.............x.x..x.x..x....x.........x..x.x.......xx.xxx....x..x..xx........x.x.................xx.........x.x..x..x...x.........................x........... [ 66%]
.............................x...........x....................................x........xx............xx.........x.........xx.x.............x.x............x...x..........x............x...xxx.x... [ 88%]
.........x.xxx.x....x...xx.......xs...x.....xs.......s....x.....x.x.........s..x..x.x..s.ssxs..x.x.s..                                                                                             [100%]
======================================================================== 717 passed, 10 skipped, 151 xfailed in 139.34s (0:02:19) ========================================================================
```
@cpcloud it is ready for review.

**Review by cpcloud — APPROVED** [review]
LGTM! Thank you

**Pull request merged**